src/Validate.php

```diff
@@ -547,6 +548,29 @@ public static function notIn(Validator $validator, $data, $pattern, $rule, $para
         }
     }
 
+    /**
+     * contains-only:<value>(,<value>)*
+     *
+     * @param \MadeSimple\Validator\Validator $validator
+     * @param array $data
+     * @param string $pattern
+     * @param string $rule
+     * @param array  $parameters
+     */
+    public static function contains(Validator $validator, $data, $pattern, $rule, $parameters)
+    {
+        foreach (Validator::getValues($data, $pattern) as $attribute => $value) {
+            if (null === $value || empty($value)) {
+                continue;
+            }
+            if (is_countable($value) && count($parameters) == count(array_intersect($value, $parameters))) {
+                continue;
+            }
+
+            $validator->addError($attribute, $rule, [':values' => implode(', ', $parameters)]);
+        }
+    }
```

Comment:
Please could you add an integration test to cover this new validation?

See `tests/Integration/ValidateContainsOnlyTest.php`  as an example.